### PR TITLE
Updated documentation and code

### PR DESCRIPTION
**Pull request by djgagne**
I added more descriptive documentation to the doc folder about setting up a new ModelGrid subclass as well as more information about the evaluation metrics and how they work.

### CODE REVIEW

**Comment by djgagne** [comment]
I updated the hagelslag code to ensure that the objects are the same with and without patches. The key change is adding a new flag to the hsdata config options called "segment_full_object". If segment_full_object is True, then the full object extent will be extracted into an STObject even if the patch_radius is specified. This should ensure that the same objects will be segmented in the scenario with and without patches because the size of the patch is filtered out from the border of the model domain. 

I have also changed the geojson writing procedure to only save out the object bounds to the geojson file. It also saves all objects from a single ensemble member run to a single geojson file instead of a separate file for each object.

**Comment by djgagne** [comment]
I just added functionality to extract patches from STObjects after tracking them in hsdata. This new functionality should ensure that the polygons output in the geojson file are in the same order as the objects in the csv and netCDF files.

File: hagelslag/processing/STObject.py [hunk 1]
```diff
@@ -101,6 +101,92 @@ def center_of_mass(self, time):
...
+                if obj_slice_local[0].stop - self.timesteps[t].shape[0] > 0 else 0
+            pad_j_l = abs(obj_slice_local[1].start) if obj_slice_local[1].start < 0 else 0
+            pad_j_u = obj_slice_local[1].stop - self.timesteps[t].shape[1] \
+                if obj_slice_local[1].stop - self.timesteps[t].shape[1] > 0 else 0
+            if obj_slice_local[0].start < 0:
+                obj_slice_local[0].start = 0
+                obj_slice_local[0].stop += pad_i_l
+            if obj_slice_local[1].start < 0:
+                obj_slice_local[1].start = 0
+                obj_slice_local[1].stop += pad_j_l
```
Comment (charlie-becker) [inline hagelslag/processing/STObject.py]: Slice object attributes appear to be read only. So it looks like you might need to fully redefine the slice or define it after the conditional.

Comment (charlie-becker) [inline hagelslag/processing/STObject.py]: Slice object attributes appear to be read only. So it looks like you might need to fully redefine the slice or define it after the conditional.

File: bin/hsdata [hunk 2]
```diff
@@ -191,18 +203,22 @@ def process_ensemble_member(run_date, member, config):
...
             elif len(model_tracks) > 0:
                 print(run_date, member, "Make Forecast Track Data")
                 forecast_data = make_forecast_track_data(model_tracks, run_date, member, config,
                                                          track_proc.model_grid.proj)
                 if patch_radius is not None:
                     print(run_date, member, "Track Data to netCDF")
-                    forecast_track_patches_to_netcdf(model_tracks, patch_radius, run_date, member, config)
+                    if model_patches is None:
+                        model_patches = model_tracks
+                    forecast_track_patches_to_netcdf(model_patches, patch_radius, run_date, member, config)
```
Comment (charlie-becker) [inline bin/hsdata:220]: The model patches do not seem to contain the model attributes which causes it to fail when converting to netCDF.

**Comment by charlie-becker** [comment]
Code looks to be working order. I verified that JSON objects are complete and in correct locations, as well as the patches themselves, and all are ordered identically.

**Comment by charlie-becker** [comment]
I also removed the grib2io requirement from requirements.txt to be inline with the environment.yml file and not throw an error when installing from scratch.

**Pull request merged**